NebulaModel/Utils/NativeInterop.cs

```diff
@@ -17,5 +19,50 @@ public static void HideWindow()
         {
             ShowWindow(GetActiveWindow(), SW_HIDE);
         }
+
+
+        [DllImport("Kernel32")]
+        private static extern bool SetConsoleCtrlHandler(EventHandler handler, bool add);
+
+        private delegate bool EventHandler(CtrlType sig);
```

Comment:
EventHandler is a very misleading name for this delegate, there's an EventHandler delegate in the BCL.

Maybe name it CtrlHandler or something.